app/controllers/profiles_controller.rb

```diff
@@ -4,6 +4,18 @@ class ProfilesController < ApplicationController
   def show
   end
   
+  def delete_image
+    images=@profile.avatars
+    if params[:index_tag].to_i == 0
+      images=[]
+    else
+      images.delete_at(params[:index_tag].to_i)
+    end
+    @profile.avatars=images
+    @profile.save!
+    redirect_to :back
```

Comment:
This is a pretty good effort. As far as I can see though this won't delete the underlying file off the filesystem, it will just 'forget' about it. This might be good enough for now but in a production environment we'd end up with lots of orphaned files with this approach.